app/App.js

```diff
@@ -0,0 +1,298 @@
+import { Component } from "react";
+import CryptoJS from "crypto-js";
+
+import CookieKitPopup from "./component/CookieKitPopup";
+import {
+  xcoobeeCookiesKey,
+  animations,
+  tokenKey,
+  euCountries,
+  cookieTypes,
+  requiredFields,
+  consentStatuses,
+  expirationTime,
+} from "./utils";
+import graphQLRequest from "./utils/graphql";
+
+export default class App extends Component {
+  // Remove cookies preferences and auth token from local storage (for easier testing)
+  static refresh() {
+    localStorage.removeItem(tokenKey);
+    localStorage.removeItem(xcoobeeCookiesKey);
+    window.location.reload();
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      animation: animations.noAnimation,
+      countryCode: "US",
+      isOffline: !XcooBee.kit.config.campaignReference,
+      isOpen: false,
+      isShown: true,
+      loading: true,
+      pulsing: false,
+      userOptions: [],
+      crowdAI: false,
+    };
+
+    this.timer = null;
+    this.errors = false;
+
+    this.checkRequiredFields();
+
+    const timestamp = localStorage[xcoobeeCookiesKey] ? JSON.parse(localStorage[xcoobeeCookiesKey]).timestamp : null;
+
+    if (timestamp && ((Date.now() - timestamp) < expirationTime)) {
+      this.fetchLocation(false, false, true);
+    } else {
+      this.fetchUserSettings();
+    }
+  }
+
+  setAnimation(countryCode, crowdAI, userSettings, savedPreferences) {
+    const { userOptions } = this.state;
+    const { config } = XcooBee.kit;
+
+    if (savedPreferences) {
+      config.cookies.forEach((cookie) => {
+        const cookieType = cookieTypes.find(type => type.key === cookie.type);
+
+        cookie.checked = JSON.parse(localStorage[xcoobeeCookiesKey]).cookies[cookieType.id];
+      });
+
+      return this.startPulsing(animations.userSettings);
+    }
+    if (userSettings && localStorage[tokenKey]) {
+      return this.startPulsing(animations.userSettings);
+    }
+    if (crowdAI && localStorage[tokenKey]) {
+      return this.startPulsing(animations.crowdIntelligence);
+    }
+    if (userOptions.length && localStorage[tokenKey]) {
+      config.cookies.forEach((cookie) => {
+        if (userOptions.includes(cookie.type)) {
+          cookie.checked = true;
+        }
+      });
+
+      return this.startPulsing(animations.userPreference);
+    }
+    if (!euCountries.includes(countryCode) && config.displayOnlyForEU) {
+      config.cookies.forEach((cookie) => {
+        if (config.checkByDefaultTypes.includes(cookie.type)) {
+          cookie.checked = true;
+        }
+      });
+
+      return this.startPulsing(animations.companyPreference);
+    }
+    this.setState({ isOpen: true });
+    return animations.noAnimation;
+  }
+
+  checkRequiredFields() {
```

Comment:
@dhurlburtusa maybe it's better to add this check to existing `initialize` method of the `CookieKit` class?

Reply:
Yes, it definitely belongs somewhere in the `CookieKit` class. After looking over the code again, `checkRequiredFields` ends up getting called when `CookieKit#initialize` is called since it calls `ReactDOM.render(<App />, ...)` which calls the `App` constructor which calls `checkRequiredFields`. So, yes, I agree the code would be better moved to `initialize` before `ReactDOM.render` is called.

However, this makes me wonder what `setParams` is useful for. Would you please explain?

Reply:
@dhurlburtusa 
if setParam() is the implementation of this: 
https://github.com/XcooBee/xcoobee-cookie-kit/#setparamparameter-value

then it is used when non required param values are set at a different time from initialize() function.  For instance when handler functions changes, or termsUrl changes. This allows more flexibility in code implementation.